In [8]:
# 1. Importar librerías necesarias
import joblib
import pandas as pd
from sklearn.metrics import classification_report, roc_auc_score
from sklearn.metrics import classification_report, roc_auc_score

In [9]:
# Cargar el modelo desde la carpeta actual
model_path = "best_rf_model.pkl"
modelo_cargado = joblib.load(model_path)
print("Modelo cargado con éxito desde la carpeta actual.")

Modelo cargado con éxito desde la carpeta actual.


In [10]:
# Cargar los datos de prueba
X_test = pd.read_csv("data/X_test.csv")
y_test = pd.read_csv("data/y_test.csv").squeeze()  # Convertir a Serie para usar como etiqueta

In [11]:
# Hacer predicciones con el modelo cargado
y_test_pred = modelo_cargado.predict(X_test)
y_test_pred_proba = modelo_cargado.predict_proba(X_test)[:, 1]

In [13]:
print("Reporte de Clasificación:")
print(classification_report(y_test, y_test_pred))

Reporte de Clasificación:
              precision    recall  f1-score   support

           0       1.00      1.00      1.00     29998
           1       1.00      0.97      0.98       497

    accuracy                           1.00     30495
   macro avg       1.00      0.98      0.99     30495
weighted avg       1.00      1.00      1.00     30495



In [14]:
print("ROC-AUC Score en Test:", roc_auc_score(y_test, y_test_pred_proba))

ROC-AUC Score en Test: 0.9828486889065575


### Conclusiones:

- Se probó el modelo final en el conjunto de datos de prueba. Los resultados obtenidos confirman el excelente rendimiento del modelo para la predicción de averías.

### Métricas de Clasificación:
- **Precisión general:** 100% para la clase 0 (sin avería) y 97% para la clase 1 (avería).
- **Recall:** El modelo detectó correctamente el 97% de las averías reales.
- **F1-Score:** Excelente balance entre precisión y recall, con un valor promedio ponderado de 0.99.
- **Soporte:** Las métricas se basan en un total de 30,495 observaciones (29,998 sin averías y 497 con averías).

### Reporte de Clasificación:
 - La precisión perfecta en la clase 0 asegura que el modelo no genera falsos positivos, evitando inspecciones innecesarias.
 - Aunque el recall de la clase 1 es alto, los falsos negativos (~3%) representan un área de mejora crítica, ya que implican averías no detectadas.

### Roc-Auc:
 - La curva ROC demuestra la capacidad del modelo para asignar correctamente probabilidades altas a las observaciones con averías (1) y bajas a las observaciones sin averías (0).
 - El área bajo la curva (AUC) cercano a 1 refleja un modelo con un excelente rendimiento.